manifests/profile/cache_gems.pp

```diff
@@ -93,38 +89,42 @@
-  bootstrap::gem { 'parslet':                        version => '1.8.0'  }
-  bootstrap::gem { 'pkg-config':                     version => '1.1.7'  }
-  bootstrap::gem { 'public_suffix':                  version => '2.0.5'  }
-  bootstrap::gem { 'rack-contrib':                   version => '1.4.0'  }
-  bootstrap::gem { 'redcarpet':                      version => '3.4.0'  }
-  bootstrap::gem { 'ruby-dbus':                      version => '0.13.0' }
-  bootstrap::gem { 'showoff':                        version => '0.18.1' }
-  bootstrap::gem { 'sinatra-websocket':              version => '0.3.1'  }
-  bootstrap::gem { 'thin':                           version => '1.7.0'  }
+  if $puppetfactory {
+    # Sinatra and Puppetfactory gems
+    bootstrap::gem { 'abalone':                        version => '0.4.1'  }
+    bootstrap::gem { 'rack':                           version => '1.6.8'  }
+    bootstrap::gem { 'rack-protection':                version => '1.5.3'  }
+    bootstrap::gem { 'rest-client':                    version => '1.8.0'  }
+    bootstrap::gem { 'sinatra':                        version => '1.4.8'  }
+    bootstrap::gem { 'tilt':                           version => '2.0.7'  }
+    bootstrap::gem { 'puppetclassify':                 version => '0.1.7'  }
+  }
+
+  if $showoff {
+    # Showoff dependencies
+    bootstrap::gem { 'addressable':                    version => '2.5.1'  }
+    bootstrap::gem { 'commonmarker':                   version => '0.14.2' }
+    bootstrap::gem { 'daemons':                        version => '1.2.4'  }
+    bootstrap::gem { 'em-websocket':                   version => '0.3.8'  }
+    bootstrap::gem { 'eventmachine':                   version => '1.2.3' }
+    bootstrap::gem { 'fidget':                         version => '0.0.4'  }
+    bootstrap::gem { 'git-version-bump':               version => '0.15' }
+    bootstrap::gem { 'gli':                            version => '2.16.0' }
+    bootstrap::gem { 'htmlentities':                   version => '4.3.4'  }
+    bootstrap::gem { 'i18n': }
+    bootstrap::gem { 'iso-639': }
+    bootstrap::gem { 'mini_portile2':                  version => '2.1.0'  }
+    bootstrap::gem { 'nokogiri':                       version => '1.6.8.1' }
+    bootstrap::gem { 'parslet':                        version => '1.8.0'  }
+    bootstrap::gem { 'pkg-config':                     version => '1.1.7'  }
+    bootstrap::gem { 'public_suffix':                  version => '2.0.5'  }
+    bootstrap::gem { 'rack-contrib':                   version => '1.4.0'  }
+    bootstrap::gem { 'redcarpet':                      version => '3.4.0'  }
+    bootstrap::gem { 'ruby-dbus':                      version => '0.13.0' }
+    bootstrap::gem { 'showoff':                        version => '0.18.1' }
+    bootstrap::gem { 'sinatra-websocket':              version => '0.3.1'  }
+    bootstrap::gem { 'thin':                           version => '1.7.0'  }
```

Comment:
There are a few duplicate packages here. Did you test combos of `$pupeptfactory`, `$showoff`, and `$learning_vm`? You should have a conditional that rejects invalid combinations.